xclim/indices/generic.py

```diff
@@ -1478,3 +1478,70 @@ def detrend(
         return ds - trend
+
+
+@declare_relative_units(thresh="<data>")
+def thresholded_events(
+    data: xr.DataArray,
+    thresh: Quantified,
+    op: str,
+    window: int,
+    thresh_stop: Quantified | None = None,
+    op_stop: str | None = None,
+    window_stop: int | None = None,
+    freq: str | None = None,
+) -> xr.Dataset:
+    r"""Thresholded events.
+
+    Finds all events along the time dimension. An event starts if the start condition is fulfilled for a given number of consecutive time steps.
+    It ends when the end condition is fulfilled for a given number of consecutive time steps.
+
+    Conditions are simple comparison of the data with a threshold: ``cond = data op thresh``.
+    The end conditions defaults to the negation of the start condition.
+
+    The resulting ``event`` dimension always has its maximal possible size : ``data.size / (window + window_stop)``.
+
+    Parameters
+    ----------
+    data : xr.DataArray
+        Variable.
+    thresh : Quantified
+        Threshold defining the event.
+    op : {">", "gt", "<", "lt", ">=", "ge", "<=", "le", "==", "eq", "!=", "ne"}
+        Logical operator defining the event, e.g. arr > thresh.
+    window: int
+        Number of time steps where the event condition must be true to start an event.
+    thresh_stop : Quantified, optional
+        Threshold defining the end of an event. Defaults to `thresh`.
+    op_stop : {">", "gt", "<", "lt", ">=", "ge", "<=", "le", "==", "eq", "!=", "ne"}, optional
+        Logical operator for the end of an event. Defaults to the opposite of `op`.
+    window_stop: int, optional
+        Number of time steps where the end condition must be true to end an event. Defaults to `window`.
+    freq: str, optional
+        A frequency to divide the data into periods. If absent, the output has not time dimension.
+        If given, the events are searched within in each resample period independently.
+
+    Returns
+    -------
+    xr.Dataset, same shape as the data except the time dimension is replaced by an "event" dimension.
+        event_length: The number of time steps in each event
+        event_effective_length: The number of time steps of even event where the start condition is true.
+        event_sum: The sum within each event, only considering the steps where start condition is true.
+        event_start: The datetime of the start of the run.
+    """
+    # condition to respect for `window_start` time steps to start a run
+    window_stop = window_stop or window
+    thresh = convert_units_to(thresh, data)
+
+    # Start and end conditions
+    da_start = compare(data, op, thresh)
+    if thresh_stop is None and op_stop is None:
+        da_stop = ~da_start
+    else:
+        thresh_stop = convert_units_to(thresh_stop or thresh, data)
+        if op_stop is not None:
+            da_stop = compare(data, op_stop, thresh_stop)
+        else:
+            da_stop = ~compare(data, op, thresh_stop)
+
```

Comment:
missing a "~" right? 
```suggestion
        if op_stop is not None:
            da_stop = ~compare(data, op_stop, thresh_stop)
        else:
            da_stop = ~compare(data, op, thresh_stop)

```

Reply:
I was expecting `op_stop` to be the actual op here. It feels more intuitive ? 

So if you pass `op=">"`, you could pass `op_stop="<" `, for example.

Reply:
Ok, since `condition_stop` is not given as input here, I guess it makes sense to make it act on `data` rather than `~data`, but later on, `op_stop` on `~data`  so it will not be consistent